library/alloc/src/collections/linked_list.rs

```diff
@@ -967,6 +967,10 @@ impl<T> LinkedList<T> {
     /// If the closure returns false, the element will remain in the list and will not be yielded
     /// by the iterator.
     ///
+    /// If the returned `DrainFilter` is not exhausted, e.g. because it is dropped without iterating
+    /// or the iteration short-circuits, then the remaining elements will be retained.
+    /// Use `drain_filter().for_each(drop)` if you do not need the returned iterator.
```

Comment:
LinkedList doesn't have a retain method, so unlike the other collections I can't recommend that as alternative.